src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/VsContainedLanguageComponentsFactoryBase.cs

```diff
@@ -59,7 +60,7 @@ internal abstract class VsContainedLanguageComponentsFactoryBase : IVsContainedL
                     return;
                 }
              
-                myHierarchy = ProjectServices.VsHierarchy;
+                myHierarchy = (ConfiguredProjectHostObject)ProjectHostProvider.UnconfiguredProjectHostObject.ActiveIntellisenseProjectHostObject;
```

Comment:
Why the explicit cast to something very specific, and not to the type of myHierarchy?